src/main/java/bio/terra/service/dataset/flight/datadelete/DataDeletionStep.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import bio.terra.common.FlightUtils;
+import bio.terra.model.DataDeletionRequest;
+import bio.terra.model.DeleteResponseModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getDataset;
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getRequest;
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getSuffix;
+
+
+public class DataDeletionStep implements Step {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final DatasetService datasetService;
+
+    private static Logger logger = LoggerFactory.getLogger(DataDeletionStep.class);
+
+    public DataDeletionStep(BigQueryPdao bigQueryPdao, DatasetService datasetService) {
+        this.bigQueryPdao = bigQueryPdao;
+        this.datasetService = datasetService;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        Dataset dataset = getDataset(context, datasetService);
+        String suffix = getSuffix(context);
+        DataDeletionRequest dataDeletionRequest = getRequest(context);
+
+        bigQueryPdao.validateDeleteRequest(dataset, dataDeletionRequest.getTables(), suffix);
+        bigQueryPdao.applySoftDeletes(dataset, dataDeletionRequest.getTables(), suffix);
+
+        // TODO: this can be more informative, something like # rows deleted per table, or mismatched row ids
```

Comment:
Is this extra information required now? If so, this should probably have a Jira ticket reference

Reply:
I will make a ticket

- [ ] made ticket and linked it here.

Reply:
https://broadworkbench.atlassian.net/browse/DR-911